src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -249,4 +253,125 @@ public override int GetHashCode ()
     /// Gets the type metadata cache for the runtime.
     /// </summary>
     public static ITypeMetadataCache Cache => cache;
-}
\ No newline at end of file
+
+    /// <summary>
+    /// Attempt to get the Swift type metadata for the given object instance
+    /// </summary>
+    /// <typeparam name="T">The type of the object</typeparam>
+    /// <param name="obj">An object to look up</param>
+    /// <param name="result">The result of looked up type metadata</param>
+    /// <returns>true on success false otherwise</returns>
+    public static bool TryGetTypeMetadata<T>(T obj, [NotNullWhen(true)] out TypeMetadata? result)
```

Comment:
`obj` is never used?

```suggestion
    public static bool TryGetTypeMetadata<T>([NotNullWhen(true)] out TypeMetadata? result)
```